api/services/SiteCreator.js

```diff
@@ -82,6 +82,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
 const createSiteFromTemplate = ({ siteParams, user, template }) => {
   let site = Site.build(siteParams)
   site.engine = "jekyll"
+  site.defaultBranch = templateForTemplateName(template).branch
```

Comment:
That is unless the question about how this responds to a malformed config? In that case, it would fallback to the Site model's default with is `master`.